### ISSUE TITLE
503 ERROR - The Lambda function associated with the CloudFront distribution is invalid. Environment variable support for Labda@Edge 

### ISSUE THREAD

**Issue by IbrahimSam96** [aws-amplify/amplify-hosting#2084]
**Provide additional details**
I connected a SSR Next.js app repo, where amplify detected SSR pages and created SSR resources ( CloudFront, Default Edge Lambda, S3 bucket, API Edge Lambda, Image Optimization Edge Lambda ).  The error I'm receiving s because Lambda@Edge doesn't support environment variables. 
**What AWS Services are you utilizing?**
Amplify Console 
**Provide additional details e.g. code snippets**
SSR Resources 
![SSRAmplify](https://user-images.githubusercontent.com/72482204/126737753-95a520d6-de62-4565-b589-40167a3eb185.PNG)
CloudWatch Log
![CloudWatchLogs](https://user-images.githubusercontent.com/72482204/126737761-7263d168-cfeb-4cf6-a8e3-8445c6a665df.PNG)

**Comment by RealDrewKlayman** [comment]
Environment variables have been added. Check to make sure you are adding them as described here:
https://github.com/aws-amplify/amplify-console/issues/1987#issuecomment-885217940

**Comment by IbrahimSam96** [comment]
> Environment variables have been added. Check to make sure you are adding them as described here:
> [#1987 (comment)](https://github.com/aws-amplify/amplify-console/issues/1987#issuecomment-885217940)

I did update my next.config.js file and the amplify terminal fails at build and gives me this error - 

`[WARNING]: (node:1239) [DEP_WEBPACK_MODULE_UPDATE_HASH] DeprecationWarning: Module.updateHash: Use new ChunkGraph API`

**Comment by IbrahimSam96** [comment]
I'm also using Firebase Admin which works only on Server Side. 
![WebPack Error](https://user-images.githubusercontent.com/72482204/127047584-fe207c86-f6cc-4218-9533-b705f78de5f9.PNG)

**Comment by RealDrewKlayman** [comment]
I don't think that has anything to do with Environment Variables not working...
Could be related to this: https://github.com/vercel/next.js/issues/15576

**Comment by IbrahimSam96** [comment]
I navigated the discussions and still unsure about the solution is! 

 package.JSON 
````
{
  "name": "graphx",
  "version": "0.1.0",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start"
  },
  "dependencies": {
    "moment": "^2.29.1",
    "next": "^11.0.1",
    "nookies": "^2.5.2",
    "react": "^17.0.2",
    "react-bootstrap": "^1.5.2",
    "react-dom": "^17.0.2",
  }
}
```

next.config.js 

```
module.exports = {
    env: {
        PRIVATE_KEY: process.env.PRIVATE_KEY,
        CLIENT_EMAIL: process.env.CLIENT_EMAIL,
        PROJECT_ID: process.env.PROJECT_ID,
        NEXT_PUBLIC_GA_ID: process.env.NEXT_PUBLIC_GA_ID
    }
  };
```

**Comment by RealDrewKlayman** [comment]
It seems like you are in the same boat as us. 
The Module Not Found is not working correctly.

This is related with https://github.com/aws-amplify/amplify-console/issues/2098.

@IbrahimSam96

**Comment by bojidaryovchev** [comment]
yo guys, I think I found a solution. For me, the approach with ```next.config.js``` did not work, so after trying various things I saw [AWS User Guide # Access Environment Variables](https://docs.aws.amazon.com/amplify/latest/userguide/environment-variables.html#access-env-vars) and managed to get it working.

1. I went to my Amplify app Build Settings
2. I edited the App build specification ```amplify.yml``` file
3. For every Environment Variable I had added in the Amplify app Environment Variables section, I added
```- echo "SOME_VARIABLE=$SOME_VARIABLE" >> .env```
4. That's it, after that I redeployed the build and I had all my variables present
